get-all-systems-with-domains.py

```diff
@@ -0,0 +1,236 @@
+    ))
+
+    # Start pulling out the service instance information directly.
+    resource_info = {}
+    resource_info["domain_name"] = resource["name"]
+    resource_info["created_at"] = resource["created_at"].split("T")[0]
+
+    # Retrieve the service plan information.
+    service_plan_guid = resource["relationships"]["service_plan"]["data"]["guid"]
+    service_plan_output = json.loads(subprocess.check_output(["cf", "curl", CF_SERVICE_PLANS_API_URI + service_plan_guid]).decode("utf-8"))
+    resource_info["service_plan"] = service_plan_output["name"]
+
+    # Retrieve the broker information.
+    service_offering_guid = service_plan_output["relationships"]["service_offering"]["data"]["guid"]
+    service_offering_output = json.loads(subprocess.check_output(["cf", "curl", CF_SERVICE_OFFERINGS_API_URI + service_offering_guid]).decode("utf-8"))
+    resource_info["broker"] = service_offering_output["name"]
+
+    # Retrieve the space information.
+    space_guid = resource["relationships"]["space"]["data"]["guid"]
+    space_output = json.loads(subprocess.check_output(["cf", "curl", CF_SPACES_API_URI + space_guid]).decode("utf-8"))
+    resource_info["space_name"] = space_output["name"]
+
+    # Retrieve the organization information.
+    org_guid = space_output["relationships"]["organization"]["data"]["guid"]
+    org_output = json.loads(subprocess.check_output(["cf", "curl", CF_ORGANIZATIONS_API_URI + org_guid]).decode("utf-8"))
+    resource_info["org_name"] = org_output["name"]
+
+    return resource_info
+
+
+def output_instances(service_instances):
+    """
+    Outputs a list of the processed service instances in CSV format.
+    """
+
+    for instance in service_instances:
+        output = "{0},{1},{2},{3},{4},{5}".format(
+            instance["broker"],
+            instance["service_plan"],
+            instance["org_name"],
+            instance["space_name"],
+            instance["domain_name"],
+            instance["created_at"]
+        )
+
+        print(output)
+
+
+def output_split_instances(service_instances):
+    """
+    Splits a list of service instances into two separate lists, one for
+    cloud.gov domains and one for Federalist domains, and outputs the results
+    seperately.
+    """
+
+    # Split out cloud.gov-only domains.
+    cloud_instances = [
+        instance for instance in service_instances
+        if instance["org_name"] != FEDERALIST_ORG_NAME
+    ]
+
+    # Split out Federalist-only domains.
+    federalist_instances = [
+        instance for instance in service_instances
+        if instance["org_name"] == FEDERALIST_ORG_NAME
+    ]
+
+    # Print out the two separate lists.
+    print("\ncloud.gov-only domains ({0} records):\n".format(
+        len(cloud_instances)
+    ))
+    output_instances(cloud_instances)
+    print("\nFederalist-only domains ({0} records):\n".format(
+        len(federalist_instances)
+    ))
+    output_instances(federalist_instances)
```

Comment:
This makes it so this script doesn't output a CSV. To make it so this script outputs a CSV, what if we did one or both of these:
- include a column for `isfederalist`
- add a flag to exclude federalist and a flag for only federalist

Reply:
Good call - I'll think on this when making adjustments for the point below about the metadata lines.  Thanks!